TombEngine/Game/effects/effects.cpp

```diff
@@ -160,6 +160,21 @@ Particle* GetFreeParticle()
 	return spark;
 }
 
+void SetSpriteSequence(Particle& particle, GAME_OBJECT_ID objectID)
+{
+	particle.Age++;
+	if (particle.Age > particle.life)
+	{
+		particle.on = false;
+		ParticleDynamics[particle.dynamic].On = false;
+		return;
+	}
+
+	int numSprites = -Objects[objectID].nmeshes - 1;
```

Comment:
Why negative sign is used here?

Reply:
Ah, I figured out myself, it's an old legacy conventon which keeps sprite count as negative number to differentiate from meshes.